src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509ChainProcessor.cs

```diff
@@ -125,6 +128,36 @@ public SafeX509ChainHandle SafeHandle
             }
         }
 
+        private static SafeX509StoreHandle GetTrustStore(
+            X509ChainTrustMode trustMode,
+            X509Certificate2Collection customTrustStore,
+            SafeX509StackHandle untrusted,
+            SafeX509StackHandle systemTrust)
+        {
+            if (trustMode == X509ChainTrustMode.CustomRootTrust)
+            {
+                SafeX509StackHandle customTrust = Interop.Crypto.NewX509Stack();
+                foreach (X509Certificate2 cert in customTrustStore)
+                {
+                    if (cert.SubjectName.RawData.ContentsEqual(cert.IssuerName.RawData))
+                    {
+                        AddToStackAndUpRef(((OpenSslX509CertificateReader)cert.Pal).SafeHandle, customTrust);
+                    }
+                    else
+                    {
+                        AddToStackAndUpRef(((OpenSslX509CertificateReader)cert.Pal).SafeHandle, untrusted);
+                    }
+                }
+
+                Interop.Crypto.X509StackAddMultiple(untrusted, systemTrust);
+                Interop.Crypto.X509StackAddMultiple(untrusted, s_userRootStore.GetNativeCollection());
+
+                return Interop.Crypto.X509ChainNew(customTrust, Interop.Crypto.NewX509Stack());
```

Comment:
What happens to `customTrust` after this point?  Do we need to dispose of it?  Is ownership transferred?

Reply:
Same question for the result of XewX509Stack().